### PR TITLE
impr: remove draft cmo service (FPLA-1483)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-1483

### Change description ###
Removes draft cmo service and leaves Case management order service. Also begins clean up of prepareCaseManagementOrder method calls.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java [hunk 1]
```diff
@@ -247,4 +284,205 @@ void shouldReturnNullIfC2DocumentBundleIsNotPopulated() {
...
+        @Test
+        void shouldReturnCaseManagementOrderWhenFullDetailsButNoPreviousOrder() {
+            CaseData caseData = getCaseData();
+
+            assertThat(caseData.getCaseManagementOrder())
+                .isEqualToComparingFieldByField(buildOrder(schedule, recitals, createCmoDirections(), action));
+        }
+
+        @Test
+        @DisplayName("can pull directions from order in about to start callback")
```
Comment (akomisarek) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java:303]: I am not sure if it's worth mentioning anything related to about to start here, should the case data has this knowledge? It seems like it is wrong direction.

Comment (akomisarek) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java:303]: Maybe the test should be somewhere else then. (could be related to to my controller question).

Maybe we can have a controller unit test?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerAboutToSubmitTest.java [hunk 2]
```diff
@@ -156,25 +163,21 @@ void shouldRemoveOrderWhenOrderActionIsNotJudgeReview() {
...
-        assertThat(caseData.getCaseManagementOrder()).isNull();
+        assertThat(response.getData()).doesNotContainKey(CASE_MANAGEMENT_ORDER_JUDICIARY.getKey());
     }
 
     private CaseManagementOrder expectedCaseManagementOrder() {
         return CaseManagementOrder.builder()
             .orderDoc(buildFromDocument(document()))
             .id(ID)
-            .directions(emptyList())
+            .directions(wrapElements(Direction.builder().assignee(ALL_PARTIES).readOnly("No").custom("Yes").build()))
```
[the remark this answers is not in the document]
Comment (swalker125) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerAboutToSubmitTest.java:180]: happy to add, is tested in unit tests more thoroughly. All the individual directions use the same logic so let me know :)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java [hunk 3]
```diff
@@ -365,4 +367,49 @@ public String getComplianceDeadline() {
...
+
+        if (preparedOrder.isSealed() && nextHearingDateList != null) {
+            preparedOrder.setNextHearingFromDynamicElement(nextHearingDateList.getValue());
+        }
+
+        return preparedOrder;
+    }
+
+    @JsonUnwrapped
+    private Directions directionsForCaseManagementOrder;
```
[the remark this answers is not in the document]
Comment (swalker125) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java:406]: happy to move, although I guess the ideal is that case data should almost be totally abstract and just work for us

File: service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java [hunk 4]
```diff
@@ -247,4 +284,205 @@ void shouldReturnNullIfC2DocumentBundleIsNotPopulated() {
...
+        for (int i = 0; i < values.length; i++) {
+            DirectionAssignee assignee = values[i];
+            builder.append(getJsonDirectionForParty(assignee.toCaseManagementOrderDirectionField(), directionString));
+
+            if (i != values.length - 1) {
+                builder.append(",");
+            }
+        }
+
+        return builder.append("}").toString();
```
Comment (akomisarek) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java:458]: Couldn't we use some serialization here, or you are afraid that it changes something? Seems not nice to handcraft json by hand.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java [hunk 5]
```diff
@@ -365,4 +367,49 @@ public String getComplianceDeadline() {
...
+        Optional<CaseManagementOrder> order = ofNullable(caseManagementOrder);
+        Optional<LocalDate> date = ofNullable(dateOfIssue);
+        Optional<DynamicList> dateList = ofNullable(cmoHearingDateList);
+        UUID idFromDynamicList = dateList.map(DynamicList::getValueCode).orElse(null);
+        String hearingDate = dateList.map(DynamicList::getValueLabel).orElse(null);
+        String dateOfIssue = date.map(x -> formatLocalDateToString(x, DATE)).orElse(null);
+        Schedule scheduleFromOrder = order.map(CaseManagementOrder::getSchedule).orElse(null);
+        List<Element<Recital>> recitalsFromOrder = order.map(CaseManagementOrder::getRecitals).orElse(null);
+        List<Element<Direction>> orderDirections = order.map(CaseManagementOrder::getDirections).orElse(emptyList());
+        Optional<Directions> directionsForCaseManagementOrder = ofNullable(getDirectionsForCaseManagementOrder());
```
[the remark this answers is not in the document]
Comment (swalker125) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java:381]: literally about to push this :)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Directions.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.COURT;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.LOCAL_AUTHORITY;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.OTHERS;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.PARENTS_AND_RESPONDENTS;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+@Data
+@Builder
+@AllArgsConstructor
+public class Directions {
```
[the remark this answers is not in the document]
Comment (swalker125) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/Directions.java:29]: they work without you having to worry about them as a developer. I feel this is a really good improvement for our data persistance issues for hidden fields. Means I only need to worry about what actually needs to happen in the controller when I am working around the cmo area, will clean up a lot of code if standardised across sdo too

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderServiceTest.java [hunk 7]
```diff
@@ -1,150 +1,257 @@
...
-            .parentsAndRespondentsAssignee(respondent)
-            .otherPartiesAssignee(other)
-            .readOnly("No")
-            .custom("Yes")
-            .dateToBeCompletedBy(LocalDateTime.of(2099, 1, 1, 10, 0, 0))
-            .build();
+    private void initMocks() {
+        DocmosisOrder order = any(DocmosisOrder.class);
+        given(documentService.getDocumentFromDocmosisOrderTemplate(order, eq(CMO))).willReturn(DOCUMENT);
+        given(templateDataService.getTemplateData(any())).willReturn(DocmosisCaseManagementOrder.builder().build());
```
Comment (akomisarek) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderServiceTest.java:255]: I think I would prefer having this on the top, I know it's not really common to have private on the top, but I would like see easily how mine mocks were setup I think it's important.

Thoughts?

File: service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java (hunk 1 above)
Comment (swalker125) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java:303]: I think I was more just trying to explain the use cases for the tests, maybe not necessary at all? Happy to remove the display names

File: service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java (hunk 4 above)
Comment (swalker125) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java:458]: can use object mapper to convert a list of directions to a string instead, I guess we don't really care how the json is made though?

Comment (akomisarek) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java:445]: Yeah we don't, just ugly if we do it by hand, it seems bad practise but find in the test code.

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerAboutToSubmitTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (douglasrice) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerAboutToSubmitTest.java:180]: Looks like there are enough tests for this already at the unit level and at the controller levels - so I'm happy with the tests as are

**Pull request merged**